scripts/imgtool/image.py

```diff
@@ -255,16 +259,50 @@ def ecies_p256_hkdf(self, enckey, plainkey):
             format=PublicFormat.UncompressedPoint)
         return cipherkey, ciphermac, pubk
 
-    def create(self, key, enckey, dependencies=None):
+    def create(self, key, enckey, dependencies=None, sw_type=None):
         self.enckey = enckey
 
+        # Calculate the hash of the public key
+        if key is not None:
+            pub = key.get_public_bytes()
+            sha = hashlib.sha256()
+            sha.update(pub)
+            pubbytes = sha.digest()
+        else:
+            pubbytes = bytes(SHA256_DIGEST_SIZE)
+
         protected_tlv_size = 0
 
         if self.security_counter is not None:
             # Size of the security counter TLV: header ('HH') + payload ('I')
             #                                   = 4 + 4 = 8 Bytes
             protected_tlv_size += TLV_SIZE + 4
 
+        if sw_type is not None:
+            if len(sw_type) > MAX_SW_TYPE_LENGTH:
+                msg = "'{}' is too long ({} characters) for sw_type. Its " \
+                    "maximum allowed length is 12 characters.".format(
+                    sw_type, len(sw_type))
+                raise click.UsageError(msg)
+
+            image_version = (str(self.version.major) + '.'
+                          + str(self.version.minor) + '.'
+                          + str(self.version.revision))
```

Comment:
`flake8` tells me the `+` in those 2 lines should align one space after the `(` in the line above.

Reply:
Done.